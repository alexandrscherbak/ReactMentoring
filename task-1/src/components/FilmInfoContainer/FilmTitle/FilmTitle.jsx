import React from 'react';

const FilmTitle = ({title}) => (
	<div>
		<span>{title}</span>

		<style jsx>
			{`
				span {
					font-weight: bold;
					color: #919191;
				}
			`}
		</style>
	</div>
);

export default FilmTitle;
