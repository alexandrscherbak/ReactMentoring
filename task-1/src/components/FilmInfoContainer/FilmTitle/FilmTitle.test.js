import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import FilmTitle from './FilmTitle.jsx';
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });

const props = {
	title: 'title',
};

function setup() {
	return mount(<FilmTitle {...props}/>);
}

describe('FilmTitle', () => {
	test('snapshot', () => {
		const component = renderer.create(
			<FilmTitle {...props}/>,
		);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('should render self and subcomponents', () => {
		const enzymeWrapper = setup();
		expect(enzymeWrapper.find('span').text()).toBe(props.title);
	});
});
