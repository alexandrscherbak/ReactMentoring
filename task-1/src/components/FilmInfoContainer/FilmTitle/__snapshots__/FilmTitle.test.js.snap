// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmTitle snapshot 1`] = `
<div
  className="jsx-433743616"
>
  <span
    className="jsx-433743616"
  >
    title
  </span>
</div>
`;
